### PR TITLE
Add permissions check for creating networks and tables, disable menus

### LINKED ISSUE #83: Update app behavior to reflect user permissions
Including but not limited to:
- [ ] workspace listing
- [ ] action buttons
- [ ] AQL query allowances

### PR DESCRIPTION
**Pull request by JackWilb**
Closes #83

Adds a permissions check for the buttons on the network panel, and disables the add table and add network button, if the user is not a writer of the workspace.

Signed in:
image

Signed out:
image

### CODE REVIEW

File: src/components/GraphDialog.vue
Comment (AlmightyYakob) [inline src/components/GraphDialog.vue:121]: You only need `async` if you plan on calling `await` in the function
```suggestion
      get(): Promise<WorkspacePermissionsSpec | null> {
        try {
          return api.getWorkspacePermissions(this.workspace);
        } catch (error) {
          return null;
        }
      },
```

**Review by AlmightyYakob — CHANGES_REQUESTED** [review]

File: src/components/GraphDialog.vue
Comment (JackWilb) [reply src/components/GraphDialog.vue:121]: I had copied the code from here for the async computed, and you'll see it has the same issue. Since this is in an async computed block, it requires that a promise is returned. That means that if we want to remove that async keyword, we'd have to replace 

```
        } catch (error) {
          return null;
        }
```

with

```
        } catch (error) {
          return new Promise(() => null);
        }
```

Or something similar.

File: src/components/TableDialog.vue
Comment (JackWilb) [reply src/components/TableDialog.vue]: Thanks for the check here. I knew I was close, but it helps to have another set of eyes on it.

See 57b16ee

File: src/components/GraphDialog.vue
Comment (JackWilb) [reply src/components/GraphDialog.vue:121]: Do you think that's preferable to the current syntax?

**Review by AlmightyYakob — DISMISSED** [review]

**Comment by AlmightyYakob** [comment]
Actually hold off merging this, I'm still seeing some bugs

File: src/components/TableDialog.vue
Comment (AlmightyYakob) [inline src/components/TableDialog.vue:213]: I believe there's a bug here, due to destructuring the `workspace` props. This kills the reactivity, so when you switch workspaces, this isn't updated. You should do the following instead:

```suggestion
    // Check for permissions
    const hasPerms = ref(false);
    watchEffect(async () => {
      const perms = await api.getWorkspacePermissions(props.workspace);
      hasPerms.value = canUpload(store.state.userInfo, perms);
    });
```

**Review by waxlamp — COMMENTED** [review]
This looks ok to me. I'm deferring to @AlmightyYakob on a binding review for this PR. Thanks for doing this, @JackWilb!

**Review by waxlamp — COMMENTED** [review]
What's going on with that empty app_logo.png file?

**Comment by JackWilb** [comment]
> What's going on with that empty app_logo.png file?

I converted the ico to a png a while ago for something and I guess I forgot to remove it from the repo. I'll rebase it out of this PR, it shouldn't be here.

**Comment by JackWilb** [comment]
This is on hold while I make the RFC for getting a users permissions for a workspace.

**Comment by waxlamp** [comment]
> This is on hold while I make the RFC for getting a users permissions for a workspace.
> 
> See [multinet-app/multinet-rfcs#27](https://github.com/multinet-app/multinet-rfcs/pull/27)

I've switched this back to draft while we're working out the RFC and other implementation details.

**Pull request closed without merging**
